corehq/form_processor/interfaces/processor.py

```diff
@@ -97,7 +97,8 @@ def ledger_db(self):
     def acquire_lock_for_xform(self, xform_id):
         lock = self.xform_model.get_obj_lock_by_id(xform_id, timeout_seconds=15 * 60)
         try:
-            lock.acquire()
+            if not lock.acquire(blocking_timeout=5):
+                raise XFormLockError(xform_id)
```

Comment:
Now this error will be raised if the lock cannot be acquired within 5 seconds, which should cancel form processing -> make the phone try again later. This is a potentially big change, so think about the consequence carefully. I'm not sure if it will break things or not, but it seems better to give up quickly rather than blocking the submission for up to 15 minutes.

Reply:
Seems ok to me but I think at the moment the phone will retry continually without backoff (@shubham1g5 can you confirm)

Reply:
@snopoke On a failure phone retries a finite number of times (2) before it shows an error on the phone and cancels the whole form submission task. Phone only tries form submission again when user clicks sync or saves a form. 

What was the earlier behaviour here - process will just wait indefinitely to acquire the lock ? If thats the case I will support this change since mobile request will anyway timeout in ~2mins.

Reply:
> What was the earlier behaviour here - process will just wait indefinitely to acquire the lock ?

@shubham1g5 yes, that's correct.

Reply:
If long blocks are quite common (which appears to be the case), then it seems like this change will result in a bad user experience: new errors will be reported to the user because syncs will fail more often. I suppose it's also possible that they already have a bad user experience because the phone would be stuck processing a submission for 15 to 30 minutes. @shubham1g5 how long does the phone wait before request timeout (does it timeout?) when performing a form submission? Is the user able to go on with other work while the form submission is in progress?

Reply:
Actually, I think nginx will timeout before the lock is acquired, so a long block like this would result in an error on the phone anyway (maybe 504 - gateway timeout)?

Reply:
@snopoke doesn't think this will create a bad user experience.